src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,286 @@
+            break;
+        case "event":
+            currTask = parseMessageContents(message, TaskTypes.EVENT);
+            tasks.addTask(currTask);
+            storage.modifyStorage(currTask, ConfirmCodes.ADDITION, tasks);
+            returnMessage = "Alright then! I've added the task to your list:" + "\n\t" + currTask + "\n"
+                + getTaskCountMessage(tasks);
+            break;
+        default:
+            throw new DukeException("Pardon me, but I did not understand what you said.");
+        }
+        return returnMessage;
+    }
+
+    /**
+     * Parses the message contents and returns the suitable Task object.
+     *
+     * @param message the message from the User
+     * @param type    The type of the task
+     * @return Task object
+     */
+    private Task parseMessageContents(String message, TaskTypes type) throws DukeException {
+        LocalDate date;
+        LocalTime timeBegin;
+        LocalTime timeEnd;
+        String[] messageArr;
+        String description;
+        String dateString;
+        String timeBeginString;
+        String timeEndString;
+
+        switch (type) {
+        case TODO:
+            messageArr = message.split(" ", 2);
+            throwIfWrongFormat(message, messageArr, TaskTypes.TODO);
+            return new ToDo(messageArr[1]);
+        case DEADLINE:
+            String messageWithoutCommand = message.substring(message.indexOf(" ") + 1);
+
+            messageArr = messageWithoutCommand.split("/");
+            throwIfWrongFormat(message, messageArr, TaskTypes.DEADLINE);
+
+            description = messageArr[0].substring(0, messageArr[0].length() - 1); //remove last " "
+            dateString = messageArr[1].substring(3);
+            timeBeginString = messageArr.length == 3 ? messageArr[2] : null;
+            date = parseDateFromString(dateString, TaskTypes.DEADLINE);
+            if (timeBeginString == null) {
+                timeBegin = null;
+            } else {
+                timeBegin = parseTimeFromString(timeBeginString, TaskTypes.DEADLINE);
+            }
+
+            return timeBeginString == null ? new Deadline(description, date) //include time only if user included it
+                : new Deadline(description, date, timeBegin);
+        case EVENT:
+            String messageWithoutCommandEvent = message.substring(message.indexOf(" ") + 1);
+
+            messageArr = messageWithoutCommandEvent.split("/");
+            throwIfWrongFormat(message, messageArr, TaskTypes.EVENT);
+
+            description = messageArr[0].substring(0, messageArr[0].length() - 1); //remove last " "
+            dateString = messageArr[1].substring(3);
+            timeBeginString = messageArr[2];
+            timeEndString = messageArr[3];
+
+            date = parseDateFromString(dateString, TaskTypes.EVENT);
+            timeBegin = parseTimeFromString(timeBeginString, TaskTypes.EVENT);
+            timeEnd = parseTimeFromString(timeEndString, TaskTypes.EVENT);
```

Comment:
Curious what will happen if the ending time is actually before the starting time? Is there any check for the validity of the duration?

Reply:
That's a really good point! Thank you for bringing that to my attention 👍